### ISSUE TITLE
[BUG]: [ExoPlayer:Playback] Free memory reached 0, forcing garbage collection with local file above 2 MB

### ISSUE THREAD

**Issue by iphonic** [TheWidlarzGroup/react-native-video#3635]
### Version

v6 (Beta)

### What platforms are you having the problem on?

Android

### Architecture

Old architecture

### What happened?

In my application I am playing multiple tracks which gets changed when the first stops playing, it works fine when the set of tracks have audio files having size less than 2 MB, but the moment it increases the log cat starts giving `Memory reached 0` warning which is probably for number users the app is getting closed.

The application is session based which has number of Audio Track Files where max size of an Audio File can be 7MB. 

How to overcome the issue? I was trying to force use old Android Media Player by using following settings, but it doesn't work.

In settings.gradle have added following entry

```
include ':react-native-video'
project(':react-native-video').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-video/android')
```

Also in react-native.config.js have added following entry

```
 module.exports = {
  assets: ['./src/assets/fonts'],
  dependencies: {
    'react-native-video': {
      platforms: {
        android: {
          sourceDir: '../node_modules/react-native-video/android',
        },
      },
    },
  },
};
```

### Reproduction

Step to reproduce this bug are:  Very simple use any audio file having size above 2 MB, it will start throwing the warning the moment it loads.

Following is my code of initialisation

```
      <Video
          audioOnly={true}
          style={{opacity: 0.0}}
          source={currentTrack ? {uri: currentTrack.url} : null}
          onError={e => console.log('Error ', e)}
          ref={ref => {
            this.player = ref;
          }}
          controls={false}
          muted={false}
          allowsExternalPlayback={false}
          disableFocus={true}
          repeat={false}
          onEnd={this.onVideoEnd}
          paused={this.state.paused}
          playInBackground={true}
          playWhenInactive={true}
          progressUpdateInterval={
            this.state.appState === 'background' ? 2000.0 : 250.0
          }
          ignoreSilentSwitch={'ignore'}
          onProgress={({currentTime, playableDuration, seekableDuration}) => {
            var progress = currentTime / seekableDuration;
            let progressPerPart = 1.0 / tracks.length;
            var partProgress = progress * progressPerPart;
            var actualProgress =
              partProgress + progressPerPart * currentTrackIndex;
            var shortProgress = actualProgress.toFixed(2);
            actualProgress = actualProgress.toFixed(3);
            if (
              actualProgress > 0 &&
              actualProgress > this.state.lastprogress
            ) {
              this.setState({
                progress: parseFloat(actualProgress),
                lastprogress: actualProgress,
              });
            } else {
              this.setState({progress: parseFloat(actualProgress)});
            }
          }}
        />
```

**Comment by freeboub** [comment]
I have clues on how to handle this issue. I will provide a patch, @iphonic if you want to do few test by patching java code, I can give you some clues !

**Comment by iphonic** [comment]
> I have clues on how to handle this issue. I will provide a patch, @iphonic if you want to do few test by patching java code, I can give you some clues !

Yes please share the patch would try if it solves the issue. Thanks.

**Comment by freeboub** [comment]
in shouldContinueLoading here: https://github.com/react-native-video/react-native-video/blob/d3cc0d0c5a67243c9b6a37c842045e9abeca3321/android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java#L528

Can you please try to replace the content of the function by:
`return super.shouldContinueLoading(playbackPositionUs, bufferedDurationUs, playbackSpeed);`
please use the last beta release also !

**Comment by iphonic** [comment]
> in shouldContinueLoading here:
> 
> https://github.com/react-native-video/react-native-video/blob/d3cc0d0c5a67243c9b6a37c842045e9abeca3321/android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java#L528
> 
> Can you please try to replace the content of the function by: `return super.shouldContinueLoading(playbackPositionUs, bufferedDurationUs, playbackSpeed);` please use the last beta release also !

Thank you, will try. Just wanted to know one thing, is the video player doing same for iOS?

**Comment by freeboub** [comment]
@iphonic No code is not available on ios. Did you get any chance to test my proposed patch ?

**Comment by iphonic** [comment]
> @iphonic No code is not available on ios. Did you get any chance to test my proposed patch ?

Yes adding Patch did reduce the error but it was still throwing it at some point.

**Comment by freeboub** [comment]
I first fix has been merged, can you still reproduce the issue on 6.0.0-RC.2 ? 
If Yes can you please provide a sample app to reproduce the issue please ?

**Comment by freeboub** [comment]
no feedback since 3 weeks, let's close this issue
